orchagent/aclorch.cpp

```diff
@@ -2919,6 +2939,60 @@ bool AclOrch::removeAclRule(string table_id, string rule_id)
     return m_AclTables[table_oid].remove(rule_id);
 }
 
+bool AclOrch::updateAclRule(shared_ptr<AclRule> rule, string table_id, string attr_name, void *data, bool oper)
+{
+    SWSS_LOG_ENTER();
+    
+    sai_object_id_t table_oid = getTableById(table_id);
+    string attr_value;
+
+    if (table_oid == SAI_NULL_OBJECT_ID) 
+    {
+        SWSS_LOG_ERROR("Failed to update ACL rule in ACL table %s. Table doesn't exist", table_id.c_str());
+        return false;
+    }
+
+    switch(aclMatchLookup[attr_name]) 
+    {
+        case SAI_ACL_ENTRY_ATTR_FIELD_IN_PORTS:
+        {
+            sai_object_id_t port_oid = *(sai_object_id_t *)data;
+            vector<sai_object_id_t> in_ports = rule->getInPorts();
+
+            if(oper == RULE_OPER_ADD) {
+                in_ports.push_back(port_oid);
+            } else {
+                vector<sai_object_id_t> :: iterator port_iter;
+                for (auto port_iter = in_ports.begin(); port_iter != in_ports.end(); port_iter++)
+                {
+                    if(*port_iter == port_oid) {
+                        in_ports.erase(port_iter);
+                        break;
+                    }
+                }
+            }
+            
+            for (const auto& port_iter: in_ports)
+            {
+                Port p;
+                gPortsOrch->getPort(port_iter, p);
+                attr_value += p.m_alias;
+                attr_value += ',';
+            }
+            attr_value.pop_back();
+
+            rule->validateAddMatch(MATCH_IN_PORTS, attr_value);  
+            m_AclTables[table_oid].rules[rule->getId()]->update(SUBJECT_TYPE_PORT_CHANGE, &aclMatchLookup[attr_name]);
```

Comment:
I do not understand why we are using SUBJECT_TYPE_PORT_CHANGE here. What is the change event here for the port? all we need to do here is call a function to set acl rule attributes.

Reply:
ACL rule update is new function added as part of this PR. Subject type is a param used during rule update, I kept it to indicate different kind of rule updates we may be doing. Idea of passing SUBJECT_TYPE_PORT_CHANGE to indicate port mask is changing for the rule. Please let me know if i am incorrectly using subject type for this purpose.